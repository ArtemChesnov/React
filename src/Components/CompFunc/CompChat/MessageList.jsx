import React from 'react';
import shortid from 'shortid';

export const MessageList = ({ message }) => {
  return (
    <div className="message-board">
      <h1 className="message-title">Чат с Душнилой</h1>
      <div className="message-chat">
        <ul className="message-list">
          {message.map((message) => (
            <li key={shortid.generate()} className="message-item">
              <h1 className="message-autor">{message.autor}</h1>
              <p className="message-text">
                {message.value}
                <span className="message-time">{message.now}</span>
              </p>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};
