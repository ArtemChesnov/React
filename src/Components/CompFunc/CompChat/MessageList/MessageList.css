.messageList {
    background: rgb(0, 32, 36);
    background: -moz-linear-gradient(90deg, rgba(0, 32, 36, 1) 0%, rgba(0, 255, 235, 0.9444152661064426) 100%);
    background: -webkit-linear-gradient(90deg, rgba(0, 32, 36, 1) 0%, rgba(0, 255, 235, 0.9444152661064426) 100%);
    background: linear-gradient(90deg, rgba(0, 32, 36, 1) 0%, rgba(0, 255, 235, 0.9444152661064426) 100%);
    min-height: 100vh;
    display: flex;
    font-size: calc(10px + 2vmin);
    color: rgb(37, 37, 37);
    display: flex;
    justify-content: center;
    align-items: center;
}