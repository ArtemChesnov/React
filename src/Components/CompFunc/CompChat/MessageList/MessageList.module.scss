.message {
    display: flex;
    flex-direction: column;
    width: 65vw;
    justify-content: space-between;
    background: rgba(255, 255, 255, 0.1);

    &__list {
        overflow: scroll;
        overflow-x: hidden;
        height: 60vh;
        list-style-type: none;
        margin: 0;
        padding: 0;


        &::-webkit-scrollbar {

            width: 10px;
            background-color: rgba(255, 255, 255, 0.2);
            border-radius: 5em;
            box-shadow: inset 1px 1px 5px #f3faf7;
        }

        &::-webkit-scrollbar-thumb {
            background-color: rgba(255, 255, 255, 0.4);
            border-radius: 5em;
        }
    }

    &__item {
        width: 75%;
        margin: 2em 2em 2em 4em;
        padding: 1em 0;
        background: rgba(255, 255, 255, 0.2);
        color: #FFF;

        &:nth-of-type(odd) {
            position: relative;

            &.message__item::after {
                content: "";
                position: absolute;
                margin: -1.5em 1.87em;
                width: 0;
                height: 0;
                left: -40px;
                top: 30px;
                border-top: 10px solid transparent;
                border-bottom: 10px solid transparent;
                border-right: 10px solid rgba(255, 255, 255, 0.2);
            }
        }

        &:nth-of-type(even) {
            position: relative;
            margin: 2em 2em 2em 10em;

            &.message__item::after {
                content: "";
                position: absolute;
                margin: -1.5em 1.87em;
                width: 0;
                height: 0;
                right: -40px;
                top: 30px;
                border-top: 10px solid transparent;
                border-bottom: 10px solid transparent;
                border-left: 10px solid rgba(255, 255, 255, 0.2);
            }
        }
    }

    &__autor {
        font-size: 1em;
        margin: -0.25em 2em;
        font-weight: 300;
    }

    &__text {
        font-weight: 300;
        font-size: 0.7em;
        margin: 0 2.8em;
        position: relative;
    }

    &__time {
        right: 0;
        position: absolute;
    }
}