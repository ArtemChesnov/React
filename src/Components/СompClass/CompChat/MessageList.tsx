import React, { Component } from 'react';

interface Message {
  id: string;
  autor: string;
  value: string;
  now: string;
}

interface ListProps {
  messages: Message[];
}

export class MessageList extends Component<ListProps> {
  constructor(props: ListProps) {
    super(props);
  }
  render(): React.ReactNode {
    return (
      <div className="message-board">
        <h1 className="message-title">Чат с Душнилой</h1>
        <div className="message-chat">
          <ul className="message-list">
            {this.props.messages.map((message) => (
              <li key={message.id} className="message-item">
                <h1 className="message-autor">{message.autor}</h1>
                <p className="message-text">
                  {message.value}
                  <span className="message-time">{message.now}</span>
                </p>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}
