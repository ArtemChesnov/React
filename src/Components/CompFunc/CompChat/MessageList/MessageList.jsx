import React from 'react';
import './MessageList.css';

export const MessageList = ({ message }) => {
  return (
    <div className="message-board">
      <h1 className="message-title">Чат с Душнилой</h1>
      <div className="message-chat">
        <ul className="message-list">
          {message.map((message, idx) => (
            <li key={idx} className="message-item">
              <h1 className="message-autor">{message.autor}</h1>
              <p className="message-text">
                {message.value}
                <span className="message-time">{message.now}</span>
              </p>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};
