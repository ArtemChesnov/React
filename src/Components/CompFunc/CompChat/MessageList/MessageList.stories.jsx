import React from 'react';
import { MessageList } from './MessageList';

export default {
  title: 'Components/MessageList',
  component: MessageList,
};

const Template = (args) => <MessageList {...args} />;

export const TestMessage = Template.bind({});

TestMessage.args = {
  message: [
    {
      autor: 'Артём',
      value: 'Hi!',
      now: '10:10',
    },
    {
      autor: 'Душнила',
      value: 'Привет Артём, капец ты скучный...',
      now: '10:12',
    },
  ],
};
