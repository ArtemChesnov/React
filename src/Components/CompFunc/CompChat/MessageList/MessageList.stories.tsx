import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { MessageList } from './MessageList';

export default {
  title: 'Components/MessageList',
  component: MessageList,
} as ComponentMeta<typeof MessageList>;

const Template: ComponentStory<typeof MessageList> = (args) => (
  <MessageList {...args} />
);

export const TestMessage = Template.bind({});

TestMessage.args = {
  messages: [
    {
      id: '1',
      author: 'Артём',
      value: 'Hi!',
      now: '10:10',
    },
    {
      id: '2',
      author: 'Душнила',
      value: 'Привет Артём, капец ты скучный...',
      now: '10:12',
    },
  ],
};
