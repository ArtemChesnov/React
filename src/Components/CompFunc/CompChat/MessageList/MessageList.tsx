import React, { FC, useEffect, useRef } from 'react';
import './MessageList.scss';

interface Message {
  id: string;
  autor: string;
  value: string;
  now: string;
}

interface ListProps {
  messages: Message[];
}

export const MessageList: FC<ListProps> = ({ messages }) => {
  const messagesEndRef = useRef<null | HTMLElement>(null);
  const scrollToBottom = () => {
    if (messagesEndRef && messagesEndRef.current) {
      messagesEndRef.current.scrollIntoView({ behavior: 'smooth' });
    }
  };
  useEffect(scrollToBottom);
  return (
    <div className="chat-wrp">
      <ul className="chat-message-list">
        {messages.map((message) => (
          <li key={message.id} className="chat-message-item">
            <h1 className="chat-message-autor">{message.autor}</h1>
            <br />
            <p className="chat-message-text">
              {message.value}
              <span className="chat-message-time">{message.now}</span>
            </p>
          </li>
        ))}
        <span ref={() => messagesEndRef}></span>
      </ul>
    </div>
  );
};
