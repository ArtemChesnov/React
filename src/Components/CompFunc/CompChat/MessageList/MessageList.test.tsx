import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MessageList } from './MessageList';

describe('MessageList', () => {
  it('Render component', () => {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    window.HTMLElement.prototype.scrollIntoView = function () {};
    render(<MessageList messages={[]} />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<MessageList messages={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render with text', () => {
    render(
      <MessageList
        messages={[
          { id: '1', autor: 'Render', value: 'Message', now: '11-00' },
        ]}
      />
    );
    expect(screen.getAllByText(/Render/));
  });
});
