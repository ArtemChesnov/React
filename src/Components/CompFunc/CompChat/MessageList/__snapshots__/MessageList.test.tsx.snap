// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="chat-wrp"
  >
    <ul
      class="chat-message-list"
    >
      <span />
    </ul>
  </div>
</DocumentFragment>
`;
